src/mega_lehmer.rs

```diff
@@ -28,7 +32,7 @@ mod tests {
 
     #[test]
     fn test_gen() {
-        let mut lehmer = MegaLehmer::new(Felt17::new(1), Felt17::new(2));
+        let mut lehmer = MegaLehmer::new(Some(Felt17::new(1)), Felt17::new(2));
```

Comment:
both have to be an option as we can either generate or give something as input .. in my branch i already did a generator, so maybe do this also as an option type

Reply:
The multiplier cant be an option since we dont have a generator yet